src/compiler/transformers/module/module.ts

```diff
@@ -820,14 +820,24 @@ namespace ts {
             return promise;
         }
 
-        function createImportCallExpressionCommonJS(arg: Expression | undefined, containsLexicalThis: boolean): Expression {
-            // import("./blah")
+        function createImportCallExpressionCommonJS(arg: Expression | undefined): Expression {
+            // import(x)
             // emit as
-            // Promise.resolve().then(function () { return require(x); }) /*CommonJs Require*/
+            // Promise.resolve(x).then(c => require(c)) /*CommonJs Require*/
             // We have to wrap require in then callback so that require is done in asynchronously
             // if we simply do require in resolve callback in Promise constructor. We will execute the loading immediately
-            const promiseResolveCall = factory.createCallExpression(factory.createPropertyAccessExpression(factory.createIdentifier("Promise"), "resolve"), /*typeArguments*/ undefined, /*argumentsArray*/ []);
-            let requireCall: Expression = factory.createCallExpression(factory.createIdentifier("require"), /*typeArguments*/ undefined, arg ? [arg] : []);
+            // For import("StringLiteral"), we prefer to emit require("StringLiteral") so that bundlers can analyze this.
+            const isStringLiteralSpecifier = arg?.kind === SyntaxKind.StringLiteral;

```

Comment:
I thought about it—didn't think of strong arguments for anything other than string literals (and possibly string-literal-like) since they aren't statically analyzable anyway (and importing number literals seems weird). ``import(`./data/${foo}.json`)`` is also statically analyzable, but we don't have a helper for that😅